blocks/term-tree-select/index.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External Dependencies
+ */
+import { unescape as unescapeString, repeat, reduce } from 'lodash';
+
+import SelectControl from '../inspector-controls/select-control';
+
+function getSelectOptions( terms, level = 0 ) {
+	return reduce( terms, ( options, term ) => (
+		options.concat( [ {
+			value: term.id,
+			label: repeat( '\u00A0', level * 3 ) + unescapeString( term.name ),
+		} ] ).concat( getSelectOptions( term.children, level + 1 ) )
+	), [] );
+}
+
+export default function TermTreeSelect( { termsTree, label, noOptionLabel, selectedTerm, onChange } ) {
+	const options = ( noOptionLabel ? [ { value: '', label: noOptionLabel } ] : [] ).concat( getSelectOptions( termsTree ) );
```

Comment:
This is a long line and difficult to read, I would suggest breaking it down a bit.

```js
const options = (
	noOptionLabel ? 
		[ { value: '', label: noOptionLabel } ] :
		[]
).concat( getSelectOptions( termsTree ) );
```

You could also use [Lodash's `compact`](https://lodash.com/docs/4.17.4#compact) to drop falsey values:

```js
const options = compact( [
	noOptionLabel && [ { value: '', label: noOptionLabel } ],
	...getSelectOptions( termsTree ),
] );
```